src/node/interfaces.cpp

```diff
@@ -182,38 +182,51 @@ class NodeImpl : public Node
     }
     int getNumBlocks() override
     {
-        LOCK(::cs_main);
-        assert(std::addressof(::ChainActive()) == std::addressof(m_context->chainman->ActiveChain()));
-        return m_context->chainman->ActiveChain().Height();
+        if (m_context->chainman) {
+            LOCK(::cs_main);
+            assert(std::addressof(::ChainActive()) == std::addressof(m_context->chainman->ActiveChain()));
+            return m_context->chainman->ActiveChain().Height();
+        }
+        return 0;
     }
     uint256 getBestBlockHash() override
     {
-        assert(std::addressof(::ChainActive()) == std::addressof(m_context->chainman->ActiveChain()));
-        const CBlockIndex* tip = WITH_LOCK(::cs_main, return m_context->chainman->ActiveChain().Tip());
+        const CBlockIndex* tip;
+        if (m_context->chainman) {
+            LOCK(cs_main);
```

Comment:
In commit "node/interfaces: Meaningful locking and avoid potential UB" (5de3e3609ab591d88281558182efdec39b065541)

It seems like this method and the two IBD methods are the only places that are changing locks or adding "meaningful locking". Since locking changes can have unexpected pitfalls I think it'd be good to split the locking parts of this commit into a separate commit. Also would be good to expand the commit message to say what's motivating the lock changes. I'm assuming it's to satisfy some lock annotations later?